### ISSUE TITLE
Build from source on arm64 - documentation and go bug

### ISSUE THREAD

**Issue by davekempe** [Icinga/icingadb#513]
## Describe the bug

Build from source fails on arm64 with a go error.  I believe related to a go bug in 1.18.4. Trying to get icingadb going on m1 mac.
I wanted to make a ticket about building from source in general too. I found the command line in the icingadb-docker repo:

So this bug is really  a few things - 
1 what is the correct way to build from source. 
2 Will this work on arm64 or are there packages for arm64 in the works?
3 To track when the go bug is fixed (probably in 1.18.5)

```
# github.com/icinga/icingadb/pkg/com
pkg/com/bulker.go:30:16: internal compiler error: assertion failed

Please file a bug report including a short program that triggers the error.
https://go.dev/issue/new
````

## To Reproduce

Provide a link to a live example, or an unambiguous set of steps to reproduce this bug. Include configuration, logs, etc. to reproduce, if relevant.

1. go version
go version go1.18.4 linux/arm64
2. git clone https://github.com/Icinga/icingadb.git
3. cd icingadb
4. go build -ldflags '-s -w' ./cmd/icingadb
 
quits with internal compiler error.
# github.com/icinga/icingadb/pkg/com
pkg/com/bulker.go:30:16: internal compiler error: assertion failed

## Expected behavior

Icingadb binary is built, and ready to use.

## Your Environment

This is debian 11 on arm64.

Include as many relevant details about the environment you experienced the problem in

* Icinga DB version: 1.0
* Icinga 2 version: 2.11
* Operating System and version: Debian 11 on arm64 (parallels on macos, but can be baremetal soon)

## Additional context

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1015088

**Comment by julianbrost** [comment]
Regarding documentation on how to build from source: Just standard `go build` combined with the somewhat common `cmd` subdirectory in Go projects will do the job, linker flags in the docker build are just there to keep the size of the image smaller. But I do understand that you don't necessarily have to be familiar with the language just to build some software from source.

**Comment by miken32** [comment]
@julianbrost FYI this will not be fixed in 1.18 per the bug report you linked to.
